### PR TITLE
Add alias support similiar to robotframework-requests library

### PR DESCRIPTION
**Pull request by nczita**
Support multiple DB connections. 
Also:
- update tests
- update RF to >=4
- add docker-compose for local testing

### CODE REVIEW

**Comment by bhirsz** [comment]
@amochin 

This PR looks as good addition - but since a long time passed it will require some changes. Are you willing to do it @nczita ?

I will put comments on specific parts.

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ before_script:
   - mysql -e 'create database my_db_test;'
   - mysql -e "GRANT ALL PRIVILEGES ON *.* TO 'root'@'127.0.0.1';" -uroot
 script:
-#  - flake8 src/
-  - if [ $DB == 'SQLite' ]; then (cd test/ && pybot SQLite3_DB_Tests.robot); fi
-  - if [ $DB == 'Postgres' ]; then (cd test/ && pybot PostgreSQL_DB_Tests.robot); fi
-  - if [ $DB == 'MySQL' ]; then (cd test/ && pybot MySQL_DB_Tests.robot); fi
+  #  - flake8 src/
```
Comment (bhirsz) [inline .travis.yml:25]: We no longer use traving.yml -we are using github Actions. See it there https://github.com/MarketSquare/Robotframework-Database-Library/blob/master/.github/workflows/common_tests.yml

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
-#PyMySQL==0.7.4
-#psycopg2==2.6.1
-robotframework>=3.0
+#PyMySQL==1.0.2
+#psycopg2==2.9.1
+robotframework>=4.0.0
```
Comment (bhirsz) [inline requirements.txt:3]: Unless we require something from the new version of the RF, it's better to stick to lower version so more people can use library (for those who can't use RF 4.0 for some reason yet). WDYT @amochin ?

File: setup.py [hunk 3]
```diff
@@ -26,13 +26,13 @@
 
 
 version_file = join(dirname(abspath(__file__)), 'src', 'DatabaseLibrary', 'version.py')
+pkg_vars  = {}
```
Comment (bhirsz) [inline setup.py:29]: The project switched from setup.py to pyproject.toml - so changes to this file are not needed anymore.

File: src/DatabaseLibrary/connection_manager.py [hunk 4]
```diff
@@ -50,7 +50,7 @@ def connect_to_database(self, dbapiModuleName=None, dbName=None, dbUsername=None
         your database credentials.
 
         Example db.cfg file
-        | [default]
+        | [alias]
```
Comment (bhirsz) [inline src/DatabaseLibrary/connection_manager.py:53]: Nice - thanks for that, we are keeping previous file configuration while allowing to use it for multiple aliased connections.

File: requirements.txt (hunk 2 above)
Comment (amochin) [reply requirements.txt:3]: Well, this requirements file doesn't define any dependencies for library users.
It's relevant for developers only - to extent the library code and run the tests.
Or so I was using this file before.

It's the pyproject.toml file, where are dependencies specified - for installing and using the library.
Currently it doesn't limit the RF version at all.

Regarding the library development, the **requirements.txt** - the current version in the master doesn't limit the RF version as well. However, the current tests use some things from later RF versions, so actually it makes sense to put an even higher limitations - like RF 5.0 maybe, would need to check it...

File: docker-compose.yml
Comment (amochin) [inline docker-compose.yml:1]: Using docker compose for running multiple containers is a nice improvement for local test, we could consider including it.. The container params would need to be updated then, as the tests have already prepared port numbers, user credentials and db names. See the [test/readme.md](https://github.com/MarketSquare/Robotframework-Database-Library/blob/master/test/readme.md) file in the master branch.

**Comment by bhirsz** [comment]
@nczita Since it looks you're either inactive or busy at the moment, I will update the PR for you. I will create new PR using your branch as a base (thanks for that you will remain author in the git history). Thanks again for contribution!

**Comment by bhirsz** [comment]
Thanks again @nczita - we have already merged PR with alias support in #196 - I used your commits as a base and all history is retained. Closing this PR.

**Comment by nczita** [comment]
Sorry, I have been very occupied lately, but I see that you have managed to take it over - with great results! 
Thanks for taking it over and do so great job @bhirsz!

**Pull request closed without merging**